prov/shm/src/smr_ep.c

```diff
@@ -1621,7 +1239,9 @@ int smr_srx_context(struct fid_domain *domain, struct fi_rx_attr *attr,
 		smr_domain->srx = ((struct fi_peer_srx_context *) (context))->srx;
 		return FI_SUCCESS;
 	}
-	return smr_ep_srx_context(smr_domain, attr->size, rx_ep);
+	return util_ep_srx_context(&smr_domain->util_domain, attr->size,
+			SMR_IOV_LIMIT, SMR_INJECT_SIZE, &smr_update,
+			&smr_domain->util_domain.lock, rx_ep);
```

Comment:
Currently efa provider could not use util domain lock as the srx lock. Efa uses util domain lock in mr operations, which are invoked inside the progress engine.

Reply:
I may need to audit whether efa really needs that domain lock when the functions are called inside the progress engine.

Reply:
Also, it seems shm call ofi_domain_init with lock type as mutex lock... Do you need to add a check for domain_attr->threading there?